### PR TITLE
Add search command

### PR DESCRIPTION
**Pull request by logical-1985516**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/SearchCommand.java [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+                return false;
+            }
+
+            SearchCommand.SearchPersonDescriptor otherSearchPersonDescriptor =
+                    (SearchCommand.SearchPersonDescriptor) other;
+            return Objects.equals(name, otherSearchPersonDescriptor.name)
+                    && Objects.equals(phone, otherSearchPersonDescriptor.phone)
+                    && Objects.equals(email, otherSearchPersonDescriptor.email)
+                    && Objects.equals(country, otherSearchPersonDescriptor.country)
+                    && Objects.equals(comment, otherSearchPersonDescriptor.comment)
```
Comment (acekhoon) [inline src/main/java/seedu/address/logic/commands/SearchCommand.java:213]: forgot to add Objects.equals(status, otherSearchPersonDescriptor.status)??

File: src/main/java/seedu/address/logic/commands/SearchCommand.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+                    new ContainsKeywordsPredicate<>(PREFIX_COUNTRY, this.getCountry());
+            predicateList.add(countrySearch);
+            ContainsKeywordsPredicate<Status> statusSearch =
+                    new ContainsKeywordsPredicate<>(PREFIX_STATUS, this.getStatus());
+            predicateList.add(statusSearch);
+            ContainsKeywordsPredicate<Comment> commentSearch =
+                    new ContainsKeywordsPredicate<>(PREFIX_COMMENT, this.getComment());
+            predicateList.add(commentSearch);
+            ContainsKeywordsPredicate<Set<Tag>> tagSearch =
+                    new ContainsKeywordsPredicate<>(PREFIX_TAG, this.getTags());
```
Comment (acekhoon) [inline src/main/java/seedu/address/logic/commands/SearchCommand.java:191]: so if optional is empty, it matches every criteria thus returning everything is it?

File: src/main/java/seedu/address/logic/commands/SearchCommand.java [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+        @Override
+        public String toString() {
+            return new ToStringBuilder(this)
+                    .add("name", name)
+                    .add("phone", phone)
+                    .add("email", email)
+                    .add("country", country)
+                    .add("comment", comment)
+                    .add("tags", tags)
```
Comment (acekhoon) [inline src/main/java/seedu/address/logic/commands/SearchCommand.java:225]: Same for this; missing status

File: src/main/java/seedu/address/logic/parser/SearchCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            searchPersonDescriptor.setEmail(ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get()));
+        }
+        if (argMultimap.getValue(PREFIX_COUNTRY).isPresent()) {
+            searchPersonDescriptor.setCountry(ParserUtil.parseCountry(argMultimap.getValue(PREFIX_COUNTRY).get()));
+        }
+        if (argMultimap.getValue(PREFIX_STATUS).isPresent()) {
+            searchPersonDescriptor.setStatus(ParserUtil.parseStatus(argMultimap.getValue(PREFIX_STATUS).get()));
+        }
+        if (argMultimap.getValue(PREFIX_COMMENT).isPresent()) {
+            searchPersonDescriptor.setComment(new Comment(argMultimap.getValue(PREFIX_COMMENT).get()));
```
Comment (acekhoon) [inline src/main/java/seedu/address/logic/parser/SearchCommandParser.java:66]: does comment have parseComment method? is it intentional to create a new object for comment? (I think some sort of valid comment test should be done? idk)

File: src/main/java/seedu/address/model/person/ContainsKeywordsPredicate.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+            return person.getEmail().value.equals(email.orElse(person.getEmail()).value);
+        } else if (prefix.equals(PREFIX_COUNTRY)) {
+            Optional<Country> country = (Optional<Country>) keywords;
+            return person.getCountry().value.equals(country.orElse(person.getCountry()).value);
+        } else if (prefix.equals(PREFIX_STATUS)) {
+            Optional<Status> status = (Optional<Status>) keywords;
+            return person.getStatus().value.equals(status.orElse(person.getStatus()).value);
+        } else if (prefix.equals(PREFIX_COMMENT)) {
+            Optional<Comment> comment = (Optional<Comment>) keywords;
+            return person.getComment().value.contains(comment.orElse(new Comment("")).value);
```
Comment (acekhoon) [inline src/main/java/seedu/address/model/person/ContainsKeywordsPredicate.java:55]: no getCommand method for person class is it?

**Review by acekhoon — APPROVED** [review]

File: src/main/java/seedu/address/logic/commands/SearchCommand.java (hunk 1 above)
Comment (logical-1985516) [reply src/main/java/seedu/address/logic/commands/SearchCommand.java:213]: Thanks for the comment, will fix soon

File: src/main/java/seedu/address/logic/commands/SearchCommand.java (hunk 3 above)
Comment (logical-1985516) [reply src/main/java/seedu/address/logic/commands/SearchCommand.java:225]: Will fix

File: src/main/java/seedu/address/logic/parser/SearchCommandParser.java (hunk 4 above)
Comment (logical-1985516) [reply src/main/java/seedu/address/logic/parser/SearchCommandParser.java:66]: So far there is no parseComment method as there is no validation for comment. We can discuss about this later.

File: src/main/java/seedu/address/logic/commands/SearchCommand.java (hunk 2 above)
Comment (logical-1985516) [reply src/main/java/seedu/address/logic/commands/SearchCommand.java:191]: Yes (for that attribute): ContainsKeywordPredicate is designed in a way such that empty Optionals (for a particular attribute) will always return true for the test predicate method.

File: src/main/java/seedu/address/model/person/ContainsKeywordsPredicate.java (hunk 5 above)
Comment (logical-1985516) [reply src/main/java/seedu/address/model/person/ContainsKeywordsPredicate.java:55]: On further thought, this also works; in fact, it also works for Name which removes the redundant isEmpty() check. Thank for pointing this out!

**Pull request merged**